### DISCUSSION TITLE
Russian locale - wrong time creation from string date

### DISCUSSION THREAD

**Discussion by Eshelion** [moment/moment#6426] [Q&A] [answered]
**Describe the bug**
If Russian locale is set, when creating date from formats which use PM and AM, it creates date with incorrect time for PM.

**To Reproduce**
Example in:
[JSFiddle](https://jsfiddle.net/eshelion/vfb7ouk5/)

Basically execute:
`const date = moment('01/27/2022 02:48 PM', 'MM/DD/YYYY h:mm A');`

**Expected behavior**
Obtaining correct date and time (in case above - 14:48 instead of 02:48).

**Comment by mattjohnsonpint** [comment, marked as answer]
Thanks for the compact reproduction! The input contains English `AM`/`PM`, so I'd parse it with the English locale and then switch to Russian for display: `moment(input, 'MM/DD/YYYY h:mm A', 'en', true).locale('ru')`. The parsing locale must match the input language.
